fee_allocator/fee_allocator.py

```diff
@@ -19,7 +19,7 @@
 
 
 class InputFees(TypedDict):
-    chainname: int
+    chainname: Union[int, float]
```

Comment:
I don't really understand what this class is/why it is being used/etc, but if the class holds both, shouldn't it always set one to the other?  This is the kind of thing that I think abstracting is quite confusing, unless we all agree that this is how we express token values and start using it all over.

That being said if it works now great.

Reply:
it it for readability. it makes it more explicit that the it is a dict holds a chain name as the key and a value of int/float. would you rather have a dict that no type annotations like this? i guess it could also just be defined as a custom type instead of wrapping it in a class, but i thought this would make it more clear